### DISCUSSION TITLE
Force re-render of the Layout

### DISCUSSION THREAD

**Discussion by TheLexoPlexx** [vercel/next.js#49370] [App Router] [answered]
Hi there,

I've got a List of People, fetched from a Database with either a Detail-Page or a "New User"-Page to the right.
However, when creating a new user (or editing an existing one). The List of course does not update. Is there any way how I can force the layout to rerender? Example is below:

```tsx
//layout.tsx
import { Suspense, ReactNode } from "react";
import Loading from "components/loading";
import styles from "./team.module.scss"
import TeamList from "./teamList";

export default async function Layout({ children }: { children: ReactNode }) {

  return (
    <div className={styles.container}>
      <Suspense fallback={<Loading />}>
        {/* @ts-expect-error Async Server Component */}
        <TeamList />
      </Suspense>
      <div className={styles.details}>
        {children}
      </div>
    </div>
  );
}
```

```tsx
//teamList.tsx
import Link from "next/link";
import styles from "./team.module.scss"
import { BsPlus } from "react-icons/bs";
import prisma from "lib/prismadb"

export const revalidate = false;

export default async function TeamList() {

  let users = await prisma.user.findMany({
    select: {
      id: true,
      name: true,
      vorname: true,
    },
    where: {
      deleted: false,
    },
    orderBy: {
      name: "asc",
    },
  })

  return (
    <div className={styles.team}>
      <Link href="/team/new" className={styles.listentry}>
        <BsPlus />
        <span>New</span>
      </Link>
      {users.map((user) => (
        <Link href={"/team/" + user.id} key={user.id} className={styles.listentry}>
          {user.name + ", " + user.vorname}
        </Link>
      ))}
    </div>
  );
}
```

I had a look at [revalidate](https://nextjs.org/docs/app/api-reference/file-conventions/route-segment-config#revalidate) as you can still see in the teamList.tsx but this is either related to https://github.com/vercel/next.js/discussions/42290 OR this is not the intended use all along.
I also checked out [revalidatePath](https://nextjs.org/docs/app/api-reference/functions/revalidatePath) when creating a new user and tried the path of the layout and also the path of the page itself, but that did not seem to do the trick either.

I used revalidatePath like this:
```tsx
  //page.tsx
  [...]

  async function formAction(formData: FormData) {
    "use server"
    console.log("creating user", formData)

    const user = await prisma.user.create({
      data: {
        email: formData.get("mail") as string,
        password: formData.get("password") as string,
        name: formData.get("name") as string,
        vorname: formData.get("vorname") as string,
      }
    })
    revalidatePath("/team/" + user.id)

    redirect("/team/" + user.id)
  }

  [...]
```

**Comment by TheLexoPlexx** [comment, marked as answer]
This can be solved (sort of) by using a `template.tsx` instead of a `layout.tsx`

**Reply by ethanmick** [reply]
I have a very similar situation as you do, a split view with the left side being the list of objects and the right side being the detail view. The user can add or edit a user. When that happens, the left-side list does not update; it keeps the old data. I've tried revalidating the path and switching to a template, but neither works.

Did you do anything else to make this happen? When you say "sort of works," what did you mean?

**Reply by TheLexoPlexx** [reply]
It only works depending on the action you perform and I haven't really managed to find out what exactly causes the rerender, I've since resorted to doing `router.refresh()` in the places where I need it and with `useTransition()` it's non-blocking and sort of smooth as well.

**Reply by ethanmick** [reply]
Thanks for your response. My current best guess is this behavior is a bug, tracked here: https://github.com/vercel/next.js/issues/52075

**Reply by TheLexoPlexx** [reply]
Yes, I did see similar behaviour after puttung revalidatePath everywhere. Thanks for mentioning that.
